.github/workflows/pack-utilities.yml

```diff
@@ -8,9 +8,15 @@ on:
 
 jobs:
   build-and-package:
-    if: ${{ github.event.workflow_run.conclusion == 'success' && github.event.workflow_run.head_branch == 'main' && github.event.workflow_run.event != 'pull_request' }}
+    if: >
+      github.event.workflow_run.conclusion == 'success' &&
+      (
+        (github.event.workflow_run.head_branch == 'main' && github.event.workflow_run.event == 'push') ||
+        github.event.workflow_run.event == 'pull_request'
+      )    
     uses: ./.github/workflows/nuget-package-template.yml
     with:
       project_name: DfE.CoreLibs.Utilities
       project_path: src/DfE.CoreLibs.Utilities
       nuget_package_name: DfE.CoreLibs.Utilities
+      custom_suffix: ${{ github.event.workflow_run.event == 'pull_request' && '-beta' || '' }}
```

Comment:
this condition doesn't look valid

Reply:
Haha I know, it is valid, it works fine, it just doesn't look it
Meaning if it is pull request then pick up -beta if not then empty!

Reply:
Could you share with me some further examples on how that is valid syntax so I can learn a little more because my brain hurts from looking at it